services/inputhost/exthost.go

```diff
@@ -625,33 +625,54 @@ func (conn *extHost) aggregateAndSendReplies(numReplicas int) {
 
 				// Note: even if this value is negative, it is ok because we should timeout immediately
 				perMsgTimer.Reset(msgAckTimeout - elapsed)
+
+				// TODO: tighten this further, by ensuring that we received exactly one ack from each replica, i.e. not N from 1 replica
+			aggregation:
 				for i := 0; i < numReplicas; i++ {
+					stat = cherami.Status_FAILED
 					select {
 					case ack, okCh := <-resCh.appendMsgAck:
-						if !okCh || ack.GetStatus() != cherami.Status_OK {
-							stat = ack.GetStatus()
-							// error means we shutdown this extent and seal it
-							go conn.close()
+						if !okCh { // channel closed, ack has all zero values
```

Comment:
good catch.. but as you said this won't happen since nobody closes this channel..